README.md

```diff
@@ -1,81 +1,217 @@
+      ...
+    }
+    fn update(&mut self, ctx: &mut UpdateCtx, old_data: Option<&T>, data: &T, env: &Env) {
+      ...
+    }
+}
+```
+
+Druid provides a number of [basic utility and layout widgets] and it's easy to implement your own. You can also compose widgets into new widgets:
+
+```rust
+fn build_widget() -> impl Widget<u32> {
+    let mut col = Column::new();
+    for i in 0..30 {
+        let button = Button::new(format!("Button {}", i), Button::noop);
+        col.add_child(Padding::new(3.0, button), 0.0);
+    }
+    Scroll::new(col)
+}
+```
+
+### layout
+
+Druid's layout protocol is [strongly inspired by Flutter]. In their `layout` method, widgets are passed a `BoxConstraint` that provides them a minimum and maximum size for layout. Widgets are also responsible for computing appropriate constraints for their children if applicable.
+
+### data
+
+Druid uses a `Data` trait to represent [value types].
+
+This is how TKTK our application state TKTK.
+(The framework needs to know whether the data has actually changed or not, which is why `T` has a `Data` bound.)
+
+In general, you can use `derive` to generate a `Data` impl for your types.
 
-On macOS, druid requires cairo; see [gtk-rs dependencies] for installation instructions.
+```rust
+#[derive(Clone, Data)]
+struct AppState {
+    which: bool,
+    value: f64,
+}
+```
 
-You may also need to set your `PKG_CONFIG_PATH`; assuming you have installed `cairo` through homebrew, you can build with,
+### lens
+
+The `Lens` datatype gives access to a part of a larger data structure. Like `Data`, This can be derived.
+
+```rust
+#[derive(Clone, Data, Lens)]
+struct AppState {
+    which: bool,
+    value: f64,
+}
+```
+
+To use the lens, wrap your widget with LensWrap (note the conversion of CamelCase to snake_case):
+
+```rust
+LensWrap::new(WidgetThatExpectsf64::new(), lenses::app_state::value);
+```
+
+## Using druid
+
+An explicit goal of druid is to be easy to build, so please open an issue if you run into any difficulties. Druid is available on crates.io and should work as a lone dependency (it re-exports all the parts of druid-shell, piet, and kurbo that you'll need):
+
+```toml
+druid = "0.3.2"
+```
+
+Since druid is currently in fast-evolving state, you might prefer to drink from the firehose:
+
+```toml
+druid = { git = "https://github.com/xi-editor/druid.git", version = "0.4" }
+```
+
+### Platform notes
+
+#### macOS
+
+On macOS, druid requires [cairo]; if you use homebrew, `brew install cairo` should be sufficient.
 
- ```shell
-$> PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig" cargo build
- ```
 #### Linux
 
 On Linux, druid requires gtk+3; see [gtk-rs dependencies] for installation instructions.
 
-run `cargo build`
-
 ## Alternatives
 
-In addition to wrappers for mature UI toolkits (mostly C++), [conrod]
-and [azul] are interesting Rust-native efforts with promise to become usable.
+Druid is only one of many ongoing [Rust-native GUI experiments]. To mention a few:
```

Comment:
Should we link to areweguiyet? Also, I think I'd like to include an Azul link, even though it no longer seems to be under active development - it's an important contribution.